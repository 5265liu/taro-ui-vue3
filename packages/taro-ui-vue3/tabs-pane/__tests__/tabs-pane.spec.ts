import { mount } from '@vue/test-utils'
import AtTabsPane from '../src/index.vue'

const factory = (values = {}, slots = { default: [''] }) => {
  return mount(AtTabsPane, {
    slots,
    props: { ...values },
  })
}

describe('AtTabsPane Snap', () => {
  it('render initial AtTabs', () => {
    const wrapper = factory({}, { default: ['test'] })
    expect(wrapper.element).toMatchSnapshot()
  })

  it('render AtTabsPane -- props customStyle', () => {
    const wrapper = factory({ customStyle: 'color:red;' })
    expect(wrapper.element).toMatchSnapshot()
  })

  it('render AtTabsPane -- props className', () => {
    const wrapper = factory({ className: 'test' })
    expect(wrapper.element).toMatchSnapshot()
  })

  it('render AtTabsPane -- props index current', () => {
    const wrapper = factory({ index: 0, current: 0 })
    expect(wrapper.element).toMatchSnapshot()
  })

  it('render AtTabsPane -- props index current', () => {
    const wrapper = factory({ index: 1, current: 0 })
    expect(wrapper.element).toMatchSnapshot()
  })
})
