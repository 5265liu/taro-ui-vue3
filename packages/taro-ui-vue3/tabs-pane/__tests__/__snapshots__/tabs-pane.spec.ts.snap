// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AtTabsPane Snap render AtTabsPane -- props className 1`] = `
<view
  class="at-tabs-pane at-tabs-pane--active test"
/>
`;

exports[`AtTabsPane Snap render AtTabsPane -- props customStyle 1`] = `
<view
  class="at-tabs-pane at-tabs-pane--active"
  style="color: red;"
/>
`;

exports[`AtTabsPane Snap render AtTabsPane -- props index current 1`] = `
<view
  class="at-tabs-pane at-tabs-pane--active"
/>
`;

exports[`AtTabsPane Snap render AtTabsPane -- props index current 2`] = `
<view
  class="at-tabs-pane at-tabs-pane--inactive"
/>
`;

exports[`AtTabsPane Snap render initial AtTabs 1`] = `
<view
  class="at-tabs-pane at-tabs-pane--active"
>
  test
</view>
`;
